### ISSUE TITLE
Flow Job Delay Not Working

### ISSUE THREAD

**Issue by ccravens** [taskforcesh/bullmq#1036]
I am not able to delay a job submitted to a flow. I have a flow where I'm adding about 7 parents to execute one after another. One of those workers I need to delay but about 30 secs. I'm doing that with the following:

```yaml
      {
        name: 'JOB_NAME',
        queueName: 'QUEUE_NAME',
       {},
        prefix: '',
       {
           delay: 30000
       },
        children: [],
      };
```

I am able to get a delay with Jobs that are not part of a flow. And I am creating a QueueScheduler for the Queues that I would like to delay with the following:

```typescript
new QueueScheduler('QUEUE_NAME', {connection: ...})
```

According to this discussion, seems like it could be a missing feature: https://github.com/taskforcesh/bullmq/discussions/998

**Comment by neighbaa** [comment]
Firstly, thanks for the hard work on this project. I've been using the library for a couple of months and I really like it.

I required this feature in a current project and I've been playing with workarounds. I noticed that jobs added with the flow producer do respect the backoff options, so the simplest solution I found is using a custom backoff strategy combined with throwing a specific error within the processor.

I'm curious about the downsides to this approach, and I'm wondering if there is a better / simpler way, that doesn't involve manually managing an error.

In the meantime, this is my code - it does seem to achieve the desired result. (I added the original exponential backoff so I get the initial delay + backoff.)

- Worker Options
```typescript
{
  settings: {
    backoffStrategies: {
      manualDelayOnceThenExponential(attemptsMade: number, error: any, job: Job) {
        // if it's the delay error, just return the delay in options
        if (error.message === 'manual-delay') {
          return job.opts.delay || 1000;
        }

        // original exponential back off
        const attemptMinusInitialDelayFailure = attemptsMade - 1;
        const exponential =
            Math.pow(2, attemptMinusInitialDelayFailure) *
            (attemptMinusInitialDelayFailure *
              (typeof job.opts.backoff === 'number'
                  ? job.opts.backoff
                  : job.opts.backoff?.delay || 1000));
    
        return exponential;
      }
    }
  }
}
```

- Processor function
```typescript
const testDelayProcessor: Processor = async (job: Job) => {

  let step = job.data.step || 0;
  
  switch (step) {
    case 0: {
      step = 1;
      await job.update({ ...job.data, step });
      throw new Error('manual-delay');
    }
    default: {
      throw new Error('Any other error');
    }
  }
}
```

Thanks again for your dedication to this project!

**Comment by roggervalf** [comment]
Hi guys, this should be available in versión 3.2.0
